src/arrayMethodReduce.js

```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let accumulator = initialValue;
+    let currIndex = 0;
+
+    if (arguments.length < 2) {
+      accumulator = this[0];
+      currIndex = 1;
+    }
```

Comment:
Try to make such checks more obvious.
```suggestion
    if (initialValue === undefined) {
      accumulator = this[0];
      currIndex = 1;
    }
```

Reply:
got it 👌